bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteFlag.java

```diff
@@ -30,12 +30,32 @@
 public enum WriteFlag {
 
     /**
-     * Writes will be acknowledged after writing to the filesystem
+     * Bit 0 - Writes will be acknowledged after writing to the filesystem
      * but not yet been persisted to disks.
      *
      * @see ForceableHandle#force()
      */
-    DEFERRED_SYNC(0x1 << 0);
+    DEFERRED_SYNC(0x1 << 0),
+    /**
+     * Bits 1,2,3 - value 0.
```

Comment:
1) For digesttype we need 3 bits, instead of taking Bits 1, 2, 3 I would suggest you to take Bits 2, 3, 4. This gives some space for SYNC types (REGULAR_SYNC and DEFERRED_SYNC). Currently it might need just 2 values (0/1) but it is better to be safe to leave space for expansion.

2) For reserving bits for digesttype, we should declare something like 
`static final int DIGEST_TYPE_BITS                = 0x7 << 2;` By doing so we are declaring that we need three bits and it would be bits 2, 3, 4.

Reply:
1. That's a great point, will update it.
2. I am not sure what the benefit will be. Since this would require code changes, someone may still mess it up. Currently, we don't throw exceptions in `getWriteFlags()` method, silently ignoring unexpected values, otherwise I could add up some tests that would have failed if someone tried to re-use them.

Reply:
yes, we have to use this `static final int DIGEST_TYPE_BITS = 0x7 << 2;` in `getWriteFlags()` method. Basically to know the digestType we should do logical and with DIGEST_TYPE_BITS first and then do the logical and with the corresponding enum value. Basically we are making it clear in getWriteFlags method that these bits are considered to be related to DigestType.

Reply:
Yeah, I did very lame mistake with bit manipulation here. :man_facepalming: 
With this logic, everything will become `DigestType.DUMMY`. I will add a test to cover this.